pkg/config/app/kuma-cp/config.go

```diff
@@ -249,6 +251,8 @@ type GeneralConfig struct {
 	// Hostname that other components should use in order to connect to the Control Plane.
 	// Control Plane will use this value in configuration generated for dataplanes, in responses to `kumactl`, etc.
 	AdvertisedHostname string `yaml:"advertisedHostname" envconfig:"kuma_general_advertised_hostname"`
+	// DNSCacheTTL represents duration for how long Kuma CP will cache result of resolving dataplane's domain name
+	DNSCacheTTL time.Duration `yaml:"dnsCacheTTL" envconfig:"kuma_general_dns_cache_ttl"`
```

Comment:
I'm hesitant whether this should be placed in `general`, `xds` for me seems more appropriate, but I don't have strong opinion about this, so I'll let you choose